js/packages/common/src/contexts/meta/pullAuctions.ts

```diff
@@ -0,0 +1,97 @@
+import { Connection } from '@solana/web3.js';
+
+import { getMultipleAccounts } from '../accounts/getMultipleAccounts';
+import { AuctionManagerV1, AuctionManagerV2 } from '../../models';
+import { ParsedAccount } from '../accounts/types';
+import { AUCTION_ID } from '../../utils/ids';
+import { BIDDER_METADATA_LEN, BIDDER_POT_LEN } from '../../actions';
+import { getProgramAccounts } from './web3';
+import { AccountAndPubkey } from './types';
+
+type AuctionManagerByAuction = Record<
+  string,
+  ParsedAccount<AuctionManagerV1 | AuctionManagerV2>
+>;
+
+interface IPullAuctionsParams {
+  connection: Connection;
+  auctionManagersByAuction: AuctionManagerByAuction;
+}
+
+export const pullAuctions = async ({
+  connection,
+  auctionManagersByAuction,
+}: IPullAuctionsParams) => {
+  const IDS_PER_CHUNK = 100;
+  const auctionIdsChunks = [
+    ...chunks(
+      getAuctionsIdsByAuctionManager(auctionManagersByAuction),
+      IDS_PER_CHUNK,
+    ),
+  ];
+
+  const auctionsChunks = await Promise.all(
+    auctionIdsChunks.map(ids =>
+      pullAuctionsByAuctionManager({ ids, connection }),
+    ),
+  );
+
+  const bidderMetadata = await getProgramAccounts(connection, AUCTION_ID, {
```

Comment:
@kespinola it was getting overloaded by the number of requests in case of getting bidderMetadata & bidderPots for each auction. Left these requests without auction filter

Reply:
Yep I encountered the same thing. Agree this is the correct approach until auctions are loaded on demand as the user scrolls the list.